### PR TITLE
Fix minimap consoles

### LINKED ISSUE #5576: (WEB REPORT BY: fozzington REMOTE: 198.245.63.50:7777) Crew Monitor Console Bugs
-When changing z-levels, the list of (active personnel? Unsure how to properly describe it) rarely, if ever, properly updates, resulting in personnel from the first z-level still being shown despite the shift (this can be evidenced in personnel icons showing up in places that would make no rational sense, as well as a failure to properly show people who may actually exist on the z-level being shown)
- The crew monitoring console tends to break about 75% of the time after usage, possibly due to attempting to shift z-levels, thus forcing a reset of the client to re-enable its function

As it stands, the current implemented cmc makes it near-impossible to track anyone not on the main station. Perhaps that is meant to be a feature, but considering the old cmc was a lot more effective in this regard, perhaps some adjustments may be necessary

### LINKED ISSUE #5945: List of the problems the crew monitor has
- ~~Z-Level swapping is just screwed, trying it kills the monitor for you. Relog or wait some time for it to work again.~~ The menu to Z-level swap is fucking DISGUSTING, it takes several tries to even punch in the z-level, constantly unfocusing the window.
- Suffocation damage isn't blue on the map screen damage listings.
- People who are dead on the map screen are always listed as being alive when moused-over.
- Positronic Brains always look dead on the map screen.
- Text overlaps occasionally in a clusterfuck on the text screen. DOES actually happen, only with really long names usually Vox, they'll crossover to the health status and no amount of window resizing helps.
- ~~Sometimes while switching over maps between rounds, the previous map will display instead of the one you're currently on. This only happens if you do not disconnect and reconnect between map changes.~~
- Sometimes when switching Z-levels, the crew monitor mixes them up, showing say, the asteroid z-level while showing main station suit sensors.

### LINKED ISSUE #10035: Go back to NanoUI CMC/adv. sec cameras.
BYOND 510 has built in JSON.

### LINKED ISSUE #13216: Crew Monitor Console is fucking up again
it's not showing the station minimap when you click on it, like so
https://puu.sh/t7KAi/cbcd2f5bda.png

It's also doing that thing where if you try to change z leves the input won't register and goes into the chat bar instead, which was fixed some time ago but seems to be happening again

### LINKED ISSUE #13779: CMC (and other nanoUI shit) doesn't show station map
#### Description of issue

The crew monitoring computer doesn't actually show a map. Ever. Neither does the bhangometer and whatever else uses nano and shows a map.

#### Length of time in which bug has been known to occur

Since 2017 at the very least. Probably a few months more. I'm honestly surprised nobody made an issue about it yet.

#### Server revision

6235e411665e05d610547dfba2d5d7d7c83b8079

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [ ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

Yeah, it's another dupe. Couldn't find an open issue.

### PR DESCRIPTION
**Pull request by Zaers**
All the minimap consoles were broken, this fixes that.

The first issue with them was that there were two procs for generating minimaps and if they worked right they would do pretty much the same thing (neither of them worked right)(`nanomapgen_DumpImageAll()`and `generateMiniMaps()`), I melded them into `generateMiniMaps()`, making that the only proc that generates the minimap images.

The second issue with them is that `generateMiniMaps()` was not processing any turfs, this was caused by all the turfs having the `NO_MINIMAP` flag due to a flag conflict with `FPRINT`, which is set for all atoms in `code/game/atoms.dm` (see `__DEFINES/setup.dm` for the actual flag definitions), I fixed that by splitting the turf specific flags into their own variable .

And the third issue with them is that the html_interface interactive maps are nearly impossible to debug and didn't load right even with the correct minimap files so I reverted those consoles back to their original nanoUI interface and thus they're all working now (their old template files were already present in `nano/templates/`).

Additionally, now that `generateMiniMaps()` does some actual work, skipping minimap generation on roundstart would be an actual performance increase and it has been added to the mapping debug verbs to replace the 'Generate NanoUI Maps' function in case prerendering the minimaps is desired

Fixes #13779
Fixes #13216
Fixes #10035
Fixes #5945
Fixes #5576

Warn me if this PR isn't atomic enough, but I had to do all this stuff to get the consoles to work
:cl:
 * bugfix: Fixed the minimap consoles

### CODE REVIEW

**Comment by PJB3005** [comment]
Look, I want this fucking shit fixed as much as anybody else, but reverting to nano is NOT THE FIX we need.

The nano based UIs have server murdering levels of lag and there's no fixing that.

**Comment by Zaers** [comment]
well shit, I guess I could try again to see if I can make the HTML interface work, but if not then I guess it'll have to stay broken

**Comment by Shadowmech88** [comment]
If not, could you just merge the other fixes? Even if it still doesn't work, it'd be less things wrong with it, right?

**Comment by Zaers** [comment]
I'll make a new branch where I can selectively get the minimap rendering thing working but not the nanoui based consoles, I'll try another time to see if I really can't fix the map not showing up in the HTML interface maps

**Comment by 9600bauds** [comment]
Thank you very very much for looking so much into this, you have no idea how glad I am to finally see what was fucked with this thing.

**Comment by Zaers** [comment]
The thing that's bugging me is that even with the \ pointing at the right place it still doesn't show the map for the html interface, but I'll check what's up with that a bit later

File: code/game/machinery/computer/adv_camera.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	desc = "Used to access the various cameras on the station with an interactive user interface."
+	circuit = "/obj/item/weapon/circuitboard/security/advanced"
+
+/obj/item/weapon/circuitboard/security/advanced
+	name = "Circuit board (Advanced Security)"
+	build_path = /obj/machinery/computer/security/advanced
+
+/obj/machinery/computer/security/advanced/attack_hand(var/mob/user as mob)
+	if (src.z > 6)
+		user << "<span class='danger'>Unable to establish a connection: </span>You're too far away from the station!"
```
Comment (Zth--) [inline code/game/machinery/computer/adv_camera.dm:12]: we use to_chat(user, ...)

File: code/game/machinery/computer/crew.dm [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+			stat |= NOPOWER
+		else
+			icon_state = initial(icon_state)
+			stat &= ~NOPOWER
+
+
+/obj/machinery/computer/crew/Topic(href, href_list)
+	if(..()) return
+	if (src.z > 6)
+		usr << "<span class='danger'>Unable to establish a connection: </span>You're too far away from the station!"
```
Comment (Zth--) [inline code/game/machinery/computer/crew.dm:49]: Same. Use to_chat

Comment (Exxion) [reply code/game/machinery/computer/crew.dm:49]: Also check if `usr` exists
I can't think of how it wouldn't, but runtimes aren't fun

**Comment by eplgr** [comment]
Does any of this fix the constant refresh bug that basically doesn't let you use the telescience computer?
That'd be good. Really good.

File: code/game/machinery/computer/crew.dm (hunk 2 above)
Comment (9600bauds) [reply code/game/machinery/computer/crew.dm:49]: `usr` will always exists and if it somehow doesn't, this can runtime no problem

**Comment by 9600bauds** [comment]
If you commit only the `generateMiniMaps()` fixes I could help you look into the HTML interface being ass.

**Comment by Zaers** [comment]
I just got back, going to commit the fix to the minimap code as a separate PR

File: code/game/machinery/computer/crew.dm (hunk 2 above)
Comment (Exxion) [reply code/game/machinery/computer/crew.dm:49]: It's still recommended

Comment (PJB3005) [reply code/game/machinery/computer/crew.dm:49]: I disagree.

Silently eating blatant errors is always worse than a 600 MiB runtime log.

**Comment by gbasood** [comment]
rip PR

**Pull request closed without merging**